From u.kleine-koenig@pengutronix.de  Wed Jan 27 17:08:59 2010
From: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Date: Tue, 26 Jan 2010 09:35:00 +0100
Subject: Driver core: make struct platform_driver.id_table const
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: Eric Miao <eric.y.miao@gmail.com>, Vikram Dhillon <opensolarisdev@gmail.com>, linux-arm-kernel@lists.infradead.org, Eric Miao <eric.miao@marvell.com>, Ben Dooks <ben-linux@fluff.org>
Message-ID: <20100126083500.GA20792@pengutronix.de>
Content-Disposition: inline


From: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>

This fixes a warning on several pxa based machines:

	arch/arm/mach-pxa/ssp.c:475: warning: initialization discards qualifiers from pointer target type

Signed-off-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Acked-by: Vikram Dhillon <dhillonv10@gmail.com>
Acked-by: Eric Miao <eric.y.miao@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/platform.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/base/platform.c
+++ b/drivers/base/platform.c
@@ -636,7 +636,7 @@ static int platform_uevent(struct device
 }
 
 static const struct platform_device_id *platform_match_id(
-			struct platform_device_id *id,
+			const struct platform_device_id *id,
 			struct platform_device *pdev)
 {
 	while (id->name[0]) {
